sdks/python/apache_beam/runners/interactive/interactive_runner.py

```diff
@@ -137,6 +137,20 @@ def run_pipeline(self, pipeline, options):
     watch_sources(pipeline)
 
     user_pipeline = ie.current_env().user_pipeline(pipeline)
+    if user_pipeline:
+      # When the underlying_runner is a FlinkRunner instance, create a
+      # corresponding DataprocClusterManager for it if no flink_master_url
+      # is provided.
+      master_url = self._get_dataproc_cluster_master_url_if_applicable(
+          user_pipeline)
+      if master_url:
+        flink_master_option = '--flink_master={}'.format(master_url)
```

Comment:
Try to use the `view_as(FlinkOptions)` function instead of parsing it as a dict.

Reply:
Sounds good! This should resolve any issues with not knowing how the options were instantiated.

I now use `options.view_as(FlinkRunnerOptions).flink_master = master_url`, which seems to function much better. Thanks!